projects/isy-angular-widgets/src/lib/incomplete-date/incomplete-date.component.spec.ts

```diff
@@ -35,315 +42,301 @@ describe('IncompleteDateComponent', () => {
-      component.inputValue = 'xx.__.____';
-      spectator.fixture.detectChanges();
-      component.onBlur();
-      expect(input.value).toBe('');
-    });
+  it('should transform the input value on losing the focus to "" when input value contains "_"', () => {
+    component.inputValue = 'xx.__.____';
+    spectator.fixture.detectChanges();
+    component.onBlur();
+    expect(input.value).toBe('');
+  });
 
-    it('should emit an event on input change', () => {
-      const onInputSpy = spyOn(component.onInput, 'emit');
+  it('should emit an event on input change', () => {
+    const onInputSpy = spyOn(component.onInput, 'emit');
 
-      component.inputValue = 'xx.__.____';
-      spectator.fixture.detectChanges();
+    component.inputValue = 'xx.__.____';
+    spectator.fixture.detectChanges();
 
-      input.dispatchEvent(keyEvent);
-      component.onInputChange(keyEvent);
-      spectator.fixture.detectChanges();
+    input.dispatchEvent(keyEvent);
+    component.onInputChange(keyEvent);
+    spectator.fixture.detectChanges();
 
-      expect(onInputSpy).toHaveBeenCalledWith(keyEvent);
-    });
+    expect(onInputSpy).toHaveBeenCalledWith(keyEvent);
+  });
 
-    it('should autocomplete the input value from __.__.____ to xx.__.____ when entering dot', () => {
+  describe('should autocomplete the input value from __.__.____ to xx.__.____ when entering dot and ', () => {
+    it(' current cursor position is 0', () => {
       input.value = '__.__.____';
       setupEvent(keyEvent, 0, 0);
       expect(input.value).toBe('xx.__.____');
     });
 
-    it('should autocomplete the input value from __.__.____ to xx.__.____ when entering dot', () => {
+    it('current cursor position is 1', () => {
       input.value = '__.__.____';
       setupEvent(keyEvent, 1, 1);
       expect(input.value).toBe('xx.__.____');
     });
 
-    it('should autocomplete the input value from __.__.____ to xx.__.____ when entering dot', () => {
+    it('current cursor position is 2', () => {
       input.value = '__.__.____';
       setupEvent(keyEvent, 2, 2);
       expect(input.value).toBe('xx.__.____');
     });
 
-    it('should autocomplete the input value from __.__.____ to xx.xx.____ when entering dot', () => {
+    it('current cursor position is 3', () => {
       input.value = '__.__.____';
       setupEvent(keyEvent, 3, 3);
       expect(input.value).toBe('xx.xx.____');
     });
 
-    it('should autocomplete the input value from __.__.____ to xx.xx.____ when entering dot', () => {
+    it('current cursor position is 4', () => {
       input.value = '__.__.____';
       setupEvent(keyEvent, 4, 4);
       expect(input.value).toBe('xx.xx.____');
     });
 
-    it('should autocomplete the input value from __.__.____ to xx.xx.____ when entering dot', () => {
+    it('current cursor position is 5', () => {
       input.value = '__.__.____';
       setupEvent(keyEvent, 5, 5);
       expect(input.value).toBe('xx.xx.____');
     });
+  });
```

Comment:
I like the idea to reduce text messages for test output, but it's not consistent with the rest of the tests and the last three tests of the describe to not check from `__.__.____ to xx.__.____ `.
Maybe its easer to change them back like below?

Reply:
Reverted - done